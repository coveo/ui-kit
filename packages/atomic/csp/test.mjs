import {equal, notEqual} from 'node:assert';
import {fork} from 'node:child_process';
import {resolve} from 'node:path';
import {describe, it, before, after} from 'node:test';
import {launch} from 'puppeteer';

describe('CSP Smoketest', () => {
  let serverProcess;
  let browser;
  before(async () => {
    browser = await launch({headless: process.env.CI === 'true'});
    serverProcess = fork(resolve(import.meta.dirname, 'server.mjs'));
    await new Promise((resolve) => serverProcess.on('message', resolve));
  });

  after(async () => {
    serverProcess.kill();
    await browser.close();
  });

  it('should load the csp-ready page without errors', async () => {
    const page = await browser.newPage();
    const errors = [];
    page.on('console', (consoleMessage) => {
      if (consoleMessage.type() === 'error') {
        errors.push(consoleMessage);
      }
    });
    await page.goto('http://localhost:3000/examples/csp.html');
    await page.waitForNetworkIdle({waitForNetworkIdle: 1000});
    equal(errors.length, 0, 'No errors should be thrown on the page');
  });

  it('should load the non-csp-ready page with errors', async () => {
    const page = await browser.newPage();
    const errors = [];
    page.on('console', (consoleMessage) => {
      if (consoleMessage.type() === 'error') {
        errors.push(consoleMessage);
      }
    });
    await page.goto('http://localhost:3000/index.html');
    await page.waitForNetworkIdle({waitForNetworkIdle: 1000});
    notEqual(errors.length, 0, 'Errors should be thrown on the page');
  });
});
