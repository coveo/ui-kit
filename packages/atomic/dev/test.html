<!doctype html>
<html dir="ltr" lang="en">

  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0, minimum-scale=1.0, maximum-scale=5.0" />
    <title>E2E tests</title>

    <script type="module">
      import { defineCustomElements } from '@coveo/atomic/loader';
      import '@coveo/atomic/themes/accessible.css';

      defineCustomElements();
    </script>

    <style>
      ::part(ripple) {
        opacity: 10%;
      }
    </style>
  </head>

  <body></body>

</html>
