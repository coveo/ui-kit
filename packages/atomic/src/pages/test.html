<!DOCTYPE html>
<html dir="ltr" lang="en">
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0, minimum-scale=1.0, maximum-scale=5.0" />
    <title>E2E tests</title>

    <script type="module" src="/build/atomic.esm.js"></script>
    <script nomodule src="/build/atomic.js"></script>
  </head>
  <body>
    <h1>Content for Integration Test will be replaced here.</h1>
  </body>
</html>
