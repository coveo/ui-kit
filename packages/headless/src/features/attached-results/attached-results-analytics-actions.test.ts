import {buildMockRaw, buildMockResult} from '../../test';
import {buildMockInsightEngine} from '../../test/mock-engine';
import {buildMockInsightState} from '../../test/mock-insight-state';
import {buildMockSearchState} from '../../test/mock-search-state';
import {
  logCaseAttach,
  logCaseDetach,
} from './attached-results-analytics-actions';

const mockLogCaseAttach = jest.fn();
const mockLogCaseDetach = jest.fn();

jest.mock('coveo.analytics', () => {
  const mockCoveoInsightClient = jest.fn(() => ({
    disable: () => {},
    logCaseAttach: mockLogCaseAttach,
    logCaseDetach: mockLogCaseDetach,
  }));

  return {
    CoveoInsightClient: mockCoveoInsightClient,
    history: {HistoryStore: jest.fn()},
  };
});

const exampleSubject = 'example subject';
const exampleDescription = 'example description';
const exampleCaseId = '1234';
const exampleCaseNumber = '5678';

const expectedMetadata = {
  caseContext: {
    Case_Subject: exampleSubject,
    Case_Description: exampleDescription,
  },
  caseId: exampleCaseId,
  caseNumber: exampleCaseNumber,
};

const expectedDocumentInfo = {
  queryPipeline: '',
  documentUri: 'example documentUri',
  documentUriHash: 'example documentUriHash',
  collectionName: 'example collectionName',
  sourceName: 'example sourceName',
  documentPosition: 1,
  documentTitle: 'example documentTitle',
  documentUrl: 'example documentUrl',
  rankingModifier: 'example rankingModifier',
  documentAuthor: 'unknown',
};

const expectedDocumentIdentifier = {
  contentIDKey: 'permanentid',
  contentIDValue: 'example contentIDValue',
};

const resultParams = {
  title: 'example documentTitle',
  uri: 'example documentUri',
  printableUri: 'printable-uri',
  clickUri: 'example documentUrl',
  uniqueId: 'unique-id',
  excerpt: 'exceprt',
  firstSentences: 'first-sentences',
  flags: 'flags',
  rankingModifier: 'example rankingModifier',
  raw: buildMockRaw({
    urihash: 'example documentUriHash',
    source: 'example sourceName',
    collection: 'example collectionName',
    permanentid: 'example contentIDValue',
  }),
};

const testResult = buildMockResult(resultParams);

describe('logCaseAttach', () => {
  it('should log #logCaseAttach with the right payload', async () => {
    const engine = buildMockInsightEngine({
      state: buildMockInsightState({
        search: buildMockSearchState({
          results: [testResult],
        }),
        insightCaseContext: {
          caseContext: {
            Case_Subject: exampleSubject,
            Case_Description: exampleDescription,
          },
          caseId: exampleCaseId,
          caseNumber: exampleCaseNumber,
        },
      }),
    });
    await engine.dispatch(logCaseAttach(testResult));

    expect(mockLogCaseAttach).toBeCalledTimes(1);
    expect(mockLogCaseAttach.mock.calls[0][0]).toStrictEqual(
      expectedDocumentInfo
    );
    expect(mockLogCaseAttach.mock.calls[0][1]).toStrictEqual(
      expectedDocumentIdentifier
    );
    expect(mockLogCaseAttach.mock.calls[0][2]).toStrictEqual(expectedMetadata);
  });
});

describe('logCaseDetach', () => {
  it('should log #logCaseDetach with the right payload', async () => {
    const engine = buildMockInsightEngine({
      state: buildMockInsightState({
        insightCaseContext: {
          caseContext: {
            Case_Subject: exampleSubject,
            Case_Description: exampleDescription,
          },
          caseId: exampleCaseId,
          caseNumber: exampleCaseNumber,
        },
      }),
    });
    await engine.dispatch(logCaseDetach(testResult.raw.urihash));

    expect(mockLogCaseDetach).toBeCalledTimes(1);
    expect(mockLogCaseDetach.mock.calls[0][0]).toStrictEqual(
      testResult.raw.urihash
    );
    expect(mockLogCaseDetach.mock.calls[0][1]).toStrictEqual(expectedMetadata);
  });
});
