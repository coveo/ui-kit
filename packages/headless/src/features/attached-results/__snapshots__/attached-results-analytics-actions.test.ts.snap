// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`attached results analytics actions when analyticsMode is \`next\` logCaseAttach should call relay.emit properly 1`] = `
[
  "InsightPanel.ItemAction",
  {
    "action": "attach",
    "context": {
      "caseNumber": "5678",
      "targetId": "1234",
      "targetType": "Case",
    },
    "itemMetadata": {
      "author": "example author",
      "title": "example documentTitle",
      "uniqueFieldName": "permanentid",
      "uniqueFieldValue": "example contentIDValue",
      "url": "example documentUri",
    },
    "position": 1,
    "searchUid": "example searchUid",
  },
]
`;

exports[`attached results analytics actions when analyticsMode is \`next\` logCaseDetach should call relay.emit properly 1`] = `
[
  "InsightPanel.DetachItem",
  {
    "context": {
      "caseNumber": "5678",
      "targetId": "1234",
      "targetType": "Case",
    },
    "itemMetadata": {
      "author": "example author",
      "title": "example documentTitle",
      "uniqueFieldName": "permanentid",
      "uniqueFieldValue": "example contentIDValue",
      "url": "example documentUri",
    },
  },
]
`;
